lib/src/file_query_util.dart

```diff
@@ -12,12 +12,22 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-library tool.dev;
+import 'dart:io';
 
-import 'package:dart_dev/dart_dev.dart' show dev, config;
+import 'package:path/path.dart' as p;
 
-main(List<String> args) async {
-  // https://github.com/Workiva/dart_dev
-  config.format.paths = ['example/', 'lib/', 'test/', 'tool/'];
-  await dev(args);
+bool isHiddenFile(File file) {
+  final path = p.normalize(file.path);
+  return p.basename(path).startsWith('.') ||
+      p.split(path).any((segment) => segment.startsWith('.'));
 }
+
+bool isNotHiddenFile(File file) => !isHiddenFile(file);
+
+bool isDartHiddenFile(File file) {
```

Comment:
`isDartHiddenFile` and `isHiddenFile` would probably be used in almost all use-cases, right?

I don't think you'd want to codemod files inside `.dart_tool`, and you'd most likely want to skip hidden directories like `.git`/`.idea`/`.vscode`.

It might be nice if these were opt-out so that codemod authors don't accidentally forget to add them.

Reply:
^ Agreed

Reply:
Yeah, that's fair. So what do you think about an optional param on `runInteractiveCodemod` to disable the filtering-out of these hidden files?

```dart
// Runs on `.dart` files, but filters out hidden files/dirs by default
runInteractiveCodemod(
  Glob('**.dart').listSync(recursive: true).whereType<File>(),
  suggestor);

// Runs on `.dart` files that are known to be in a hidden directory, so the
// filtering of hidden files is disabled
runInteractiveCodemod(
  Glob('.dart_tool/build/generated/**.dart').listSync(recursive: true).whereType<File>(),
  suggestor,
  ignoreHiddenFiles: false);
```

Reply:
I like it!

Reply:
I guess another option would be to keep something like `FIleQuery` around just for the purpose of building file input sets more easily.. that way all of the listing/filtering logic is contained within `FileQuery` and `runInteractiveCodemod` runs on exactly what it is given (i.e. `Iterable<File>`) and doesn't do any additional filtering or require any additional params.

```dart
class FileQuery implements Set<File> {
  FileQuery.fromGlob(Glob glob, {bool ignoreHiddenFiles = true});
  FileQuery.fromIterable(Iterable<File> files);
}

// Single glob, ignores hidden files by default (common use case)
final files = FileQuery.fromGlob(Glob('**.dart', recursive: true));

// Individual files (common use case)
final files = ['pubspec.yaml', 'build.yaml'];

// Combine multiple input sets
final files = {'pubspec.yaml', 'build.yaml'}
    .union(FileQuery.fromGlob(Glob('**.dart', recursive: true)));

// Use globs with more specific exclusions
final files = FileQuery.fromGlob(Glob('**.dart', recursive: true))
    .difference(FileQuery.fromGlob(Glob('build/**', recursive: true)));

// Disable hidden file filtering for special use cases
final files = FileQuery.fromGlob(
    Glob('.dart_tool/build/generated/**.dart', recursive: true),
    ignoreHiddenFiles: false);

runInteractiveCodemod(files, suggestor, ...);
```

Thoughts?

Reply:
While I like having all of the file filtering logic decoupled from `runInteractiveCodemod`, it seems less convenient to have to have another layer around the Glob, and it's not immediately obvious what FileQuery is providing on top of Glob's functionality.

---

If we do end up moving forward with that approach,  does making FileQuery a class provide any benefit, or can we just use functions?

```dart
-final files = FileQuery.fromGlob(Glob('**.dart', recursive: true));
+final files = queryFilesFromGlob(Glob('**.dart', recursive: true));
```